src/linux_mcp_server/tools/logs.py

```diff
@@ -170,6 +173,7 @@ async def read_log_file(  # noqa: C901
 
         # For local execution, validate path
         if not host:
+            bin_path = shutil.which("tail")
```

Comment:
why not using get_bin_path here too? probably could just move the `await get_bin_path` to the top, and that would execute regardless if it is local or remote

Reply:
> We could have `get_remote_bin_path` and `get_bin_path`, but I don't want to make any more kitchen sink functions.

I understand where you're going with this one, but, it feels to me that the entrypoint of deciding if a tool will run on remote or local, is the `host`. If that's the case, `host` is being set at the tool parameter level and being passed down until we reach the `execute_command`. Seems pretty solid to me to avoid bugs, and I don't necessarily know if there is any real difference that we need to care about "I want to check `tail` locally` to "I want to check `tail` remotely" 🤔

Reply:
The nature of a remote call versus a local call is quite different. Encapsulating theme is separate functions simplifies the code and makes testing and maintenance easier.

Reply:
Ah, okay! I think I got it now.